expression/builtin_math.go

```diff
@@ -526,11 +535,12 @@ func (c *floorFunctionClass) getFunction(args []Expression, ctx context.Context)
 		return nil, errors.Trace(err)
 	}
 
-	retTp, argTp := fixFloorAndCeilType(args[0])
+	retTp, argTp := getEvalTp4FloorAndCeil(args[0])
 	bf, err = newBaseBuiltinFuncWithTp(args, ctx, retTp, argTp)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	setFlag4FloorAndCeil(&bf, args[0])
```

Comment:
we can only check whether `mysql.HasBinaryFlag(args[0].Flag)` here,
and `set bf.tp.Flag |= mysql.BinaryFlag` when it's true.

the other code seem useless?